include/deal.II/matrix_free/vector_access_internal.h

```diff
@@ -247,6 +247,12 @@ namespace internal
                             VectorizedArrayType *dof_values,
                             std::integral_constant<bool, true>) const
     {
+#ifdef DEBUG
+      for (unsigned int i = 0; i < dofs_per_cell; ++i)
+        for (unsigned int v = 0; v < VectorizedArrayType::size(); ++v)
+          vector_access(vec, dof_index + v + i * VectorizedArrayType::size());
+#endif
+
```

Comment:
This is at least surprising for someone reading the code because you load the vector twice. I would prefer to do not do the `load` part below in `debug` mode (i.e., use an `#if` / `#else` branch) or at the very least a extensive comment. The former assumes we have other paths where we test `VectorizedArray::load`.

Reply:
I have added the `#if/#else` structure and added some comments.